### ISSUE TITLE
Add SmartModule transform to backward compatibility tests

### ISSUE THREAD

**Issue by galibey** [infinyon/fluvio#3037]
Need to use smartmodules payload in backward compatibility tests to prevent regression in `transform` usages. 

Follow-up issue after #3035

**Comment by sehz** [comment]
Let's remove SmartModuleKind field altogether

**Comment by galibey** [comment]
> Let's remove SmartModuleKind field altogether

I created a ticket for it https://github.com/infinyon/fluvio/issues/3042
